docs/about/themes-translation.md

```diff
@@ -0,0 +1,147 @@
+# MkDocs themes translation
+
+How to add a locale translation to mkdocs built-in themes.
+
+---
+
+## Localization tooling prerequisites
+
+Theme localization uses the [babel][babel] project and its `pybabel`
+command-line to handle localization files generation and compilation.
+
+Make sure they are installed in your environment:
+
+```bash
+pip install -r requirements/project.txt
+```
+
+[babel]: http://babel.pocoo.org/en/latest/cmdline.html
+
+## Contributing theme translations
+
+If your favorite language locale is not supported yet by the `mkdocs` and
+`readthedocs` built-in themes, you can easily contribute it to the project by
+following the steps below.
+
+### Creating the theme localization files
+
+Theme localization is achieved by adding a `messages.po` file to the `locales`
+directory tree of the themes. This is easily achieved using the `pybabel`
+command-line which will initialize this file for you by deriving it from the
+default `en` (English) one.
```

Comment:
This is not the recommended way to do this. Babel itself [suggests][1]:

> The general procedure for building message catalogs looks something like this:
> 
> * use a tool (such as `xgettext`) to extract localizable strings from the code base and write them to a POT (PO Template) file.
> * make a copy of the POT file for a specific locale (for example, “en_US”) and start translating the messages
> * use a tool such as msgfmt to compile the locale PO file into a binary MO file
> * later, when code changes make it necessary to update the translations, you regenerate the POT file and merge the changes into the various locale-specific PO files, for example using `msgmerge`

The problem with deriving a new PO file from an existing one is that there is no automated way to update the PO files for changes in the templates. However, if the documented workflow is to generate PO files directly from the templates, then we can be sure the PO files are up-to-date. It even gives us a way to update existing PO files (generate the template and merge it with the existing file).

Looking at Babel's documentation for their [CLI][2] it would seem that after changes have been made to a template (or other files with translatable strings), we would need to run `extract` to generate a new PO template (or POT file). Then translators should be instructed to create a new locale by initializing (the `init` command) from the POT file, not the `en` PO file. Likewise, to update an existing locale, the translator can `update` from the POT file.

We have three options:

1. We can autogenerate the POT file by checking for changes to the relevant files and when appropriate rerun the `extract` command. This might make sense as a GitHub workflow on push.
2. We could require PRs which make changes to the relevant files to manually run `extract` and include the updated POT file in the PR.
3. We could instruct translators to run `extract` to generate a temporary POT file.

Option 2 is no good as it will just result in a bunch of incomplete PRs. For option 3 to work, we need to do all the legwork so that they only need to call a single simple command. And if we are going to do that, then we might as well take the extra step to have that command called automatically. Therefore, I would prefer option 1. I realize this requires more work as we need to provide babel with a configuration for doing the extraction, but I'd rather do it right the first time.

[1]: http://babel.pocoo.org/en/latest/messages.htm
[2]: http://babel.pocoo.org/en/latest/cmdline.html